Test1/ConcurrentPriorityQueue/ConcurrentPriorityQueue.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace ConcurrentPriorityQueue;
+
+using System.Collections;
+
+/// <summary>Represents a thread-safe max priority queue.</summary>
+/// <typeparam name="TElement">Specifies the type of elements in the queue.</typeparam>
+/// <typeparam name="TPriority">Specifies the type of priority associated with enqueued elements.</typeparam>
+public class ConcurrentPriorityQueue<TElement, TPriority>
+{
+    private readonly object lockObject = new();
+
+    // By the task we need to dequeue element with highest priority, but by default PriorityQueue dequeues element with the lowest one
+    // We hack that by reversing internal comparator
+    private readonly PriorityQueue<TElement, TPriority> queue =
+        new(Comparer<TPriority>.Create((a, b) => -Comparer.Default.Compare(a, b)));
+
+    /// <summary>Gets the number of elements contained in the ConcurrentPriorityQueue.</summary>
+    public int Size
+    {
+        get
+        {
+            lock (this.lockObject)
+            {
+                return this.queue.Count;
+            }
+        }
```

Comment:
```suggestion
    public int Size
    {
        get
        {
            lock (lockObject)
            {
                return queue.Count;
            }
        }
```
`this` везде можно опустить.

Reply:
`this` везде написан, чтобы соответствовать вот этому [правилу](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1101.md) StyleCop

Reply:
-0.5